packages/website/src/theme/BlogSidebar/Content/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { groupBlogSidebarItemsByYear } from '@docusaurus/plugin-content-blog/client';
+import { useThemeConfig } from '@docusaurus/theme-common';
+import type { Props } from '@theme/BlogSidebar/Content';
+import Heading from '@theme/Heading';
+import type { ReactNode } from 'react';
+import React, { memo } from 'react';
+import Markdown from 'react-markdown';
+
+import styles from './styles.module.css';
+
+function BlogSidebarYearGroup({
+  year,
+  yearGroupHeadingClassName,
+  children,
+}: {
+  year: string;
+  yearGroupHeadingClassName?: string;
+  children: ReactNode;
+}): React.JSX.Element {
+  return (
+    <div className={styles.blogSidebarContent} role="group">
+      <Heading as="h3" className={yearGroupHeadingClassName}>
+        {year}
+      </Heading>
+      {children}
+    </div>
+  );
+}
+
+function BlogSidebarContent({
+  items: itemsRaw,
+  yearGroupHeadingClassName,
+  ListComponent,
+}: Props): ReactNode {
+  const items = itemsRaw.map(item => ({
+    ...item,
+    // @ts-expect-error -- BlogSidebarItem only allows title: string, but we render as Markdown
+    title: <Markdown>{item.title}</Markdown>,
```

Comment:
This still gives a type checking error. I suspect you need to place the `@ts-expect-error` elsewhere.

But really, I would suggest you do the cast here—Docusaurus is not going to change the type of `BlogSidebarItem` so you don't actually gain any soundness.

Reply:
> Docusaurus is not going to change the type of `BlogSidebarItem`

Funny, I was thinking of filing an issue suggesting it allow any React node type for the title...

Reply:
The type of Docusaurus describes what we will pass to these components by default, not what they can accept by principle; narrower types may allow swizzlers to do other things they otherwise can't do, say, `title.toUpperCase()`. Not easy to provide types in a customizable system :)